daemon/job/active.go

```diff
@@ -132,8 +133,32 @@ func (m *modePush) Type() Type { return TypePush }
 
 func (m *modePush) PlannerPolicy() logic.PlannerPolicy { return *m.plannerPolicy }
 
-func (m *modePush) RunPeriodic(ctx context.Context, wakeUpCommon chan<- struct{}) {
-	m.snapper.Run(ctx, wakeUpCommon)
+func (m *modePush) RunPeriodic(
+	ctx context.Context, wakeUpCommon chan<- struct{},
+) {
+	if m.interval == nil {
+		m.snapper.Run(ctx, wakeUpCommon)
+		return
+	}
```

Comment:
If `m.interval != nil`, then the snapper doesn't run. I don't think this is acceptable. I think we should

1. either make them mutually exclusive, i.e., error out during config parsing (and assert/panic at runtime),
2. or make them merge, i.e., spawn two goroutines, one for `m.snapper.Run()` and one for the `m.interval` ticker that you added here.

Reply:
That's not a problem, because `m.interval != nil` only for
```yaml
    snapshotting:
      type: "manual"
```
```go
	if _, ok := in.Snapshotting.Ret.(*config.SnapshottingManual); ok {
		m.interval = in.Interval
	}
```

Reply:
That's still not great. The current version of this PR allows the user to configure 

```
interval: 20min
snapshotting:
  type: periodic
  interval: ...
```

The `snapshotting.interval` will take effect, and the top-level `interval` will be silently ignored.

I'll figure something out and push it into this PR